src/Compilers/CSharp/Test/Semantic/Semantics/NameOfTests.cs

```diff
@@ -1168,5 +1168,66 @@ public void NameofTypeParameterInParameterInitializer()
 }";
             var compilation = CreateCompilationWithMscorlib45(source).VerifyDiagnostics();
         }
+
+        [Fact, WorkItem(10467, "https://github.com/dotnet/roslyn/issues/10467")]
+        public void NameofFixedBuffer()
+        {
+            var source =
+@"
+using System;
+unsafe struct Struct1
+{
+  public fixed char MessageType[50];
+
+  public override string ToString()
+  {
+    return nameof(MessageType);
+  }
+
+  public void DoSomething(out char[] x)
+  {
+    x = new char[] {};
+    Action a = () => { System.Console.WriteLine(nameof(x)); };
```

Comment:
This scenario is applicable to VB as well. Does it work? Consider adding a test.